app/controllers/projects/violations_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Show and remove non-compliant Observations from a Project
+module Projects
+  # Actions
+  # -------
+  # edit (get)
+  # update (patch)
```

Comment:
@nimmolo, @mo-nathan 
What should be the action names? Because there was a form involved I settled on edit/update. Alternatives include `index`, or something(s) unique.

Reply:
The way I've been doing (re-doing) these, the page that shows the list of all (of the violators) would be an `index` action. 

It's debatable of course — we have usually thought of these as "edit" forms in the past, but Rails CRUD `edit` routes require an ID, and for me it feels like more of a filtered `index`. I might call this a "Project::ObservationsController".

If you have buttons or checkboxes with submit, the `form` should submit to an `update` action, though, in my opinion.

Reply:
@nimmolo: Thanks for the insightful feedback. I'll conform to how you're doing it. 
It is a kind of filtered index, but the template differs from a normal index.

Reply:
@nimmolo I'm still thinking about how to do this?
Can you point me at an example where you used index in a namespaced controller?
Also, I don't see a way around requiring an `:id` (or equivalent); the page doesn't list globally all violations,  but rather all the violations of one project.

Reply:
Oh - yes, even when you do it as an index, the CRUD route will need a `project_id`. 

(A good example of using an index with links for new/edit/create/update is the Namings table in show observation - that has its own route, Observation::NamingsController#index. If you go to /observations/:observation_id/namings, you should get the table all by itself.)

I was thinking, in the event you were making this an `#edit` action instead of an index, the Rails "resource route" would want an `observation_id` by default, because it would assume it's a page for editing a single observation belonging to a Project (like editing a Description belonging to a Name).  

In a sense maybe it is, but to me this is sort of more "bulk editing" project observations; it still seems like an index with a direct link to an update action that can handle multiple observation ids. 

Anyway i don't feel too strongly about it being an index.